sqlglot/dialects/presto.py

```diff
@@ -173,6 +173,37 @@ def _unix_to_time_sql(self: Presto.Generator, expression: exp.UnixToTime) -> str
     return f"FROM_UNIXTIME(CAST({timestamp} AS DOUBLE) / POW(10, {scale}))"
 
 
+def _jsonextract_sql(self: Presto.Generator, expression: exp.JSONExtract) -> str:
+    is_json_extract = self.dialect.settings.get("variant_extract_is_json_extract", True)
+
+    if isinstance(is_json_extract, str):
+        # if the kwarg was set by the user it's stored in a string
+        is_json_extract = is_json_extract.lower() == "true"
+
+    # Generate JSON_EXTRACT unless the user has configured that a Snowflake / Databricks
+    # VARIANT extract (e.g. col:x.y) should map to dot notation (i.e ROW access) in Presto/Trino
+    if not expression.args.get("variant_extract") or is_json_extract:
+        return self.func(
+            "JSON_EXTRACT", expression.this, expression.expression, *expression.expressions
+        )
+
+    this = self.sql(expression, "this")
+
+    # Convert the JSONPath extraction `JSON_EXTRACT(col, '$.x.y) to a ROW access col.x.y
+    segments = []
+    for path_key in expression.expression.expressions:
+        if not isinstance(path_key, exp.JSONPathKey):
+            continue
```

Comment:
When can we ever _not_ get a `JSONPathKey` here? I'm worried that `continue` is gonna silently drop information that it shouldn't.

Reply:
My thought was to rule out `exp.JSONPathRoot` (and other segment parts such as `exp.JSONPathSubscript` if they appear), as they shouldn't be transpilable to dot notation. Do we want to only remove the root and treat any other malformed input as malformed output?

Reply:
We shouldn't skip anything silently. You can do an `lstrip` to remove the root `$.`, but for other parts that we can't generate we should do `self.unsupported`, imho.

Reply:
Isn't it fine to skip `exp.JSONPathRoot` and to hit `self.unsupported` for all non path keys?

Reply:
Isn't that achieved with the above `if` block?

Reply:
Sorry if I'm misunderstanding, I mean that all JSONPaths are canonicalized with a leading `exp.JSONPathRoot`  but that shouldn't be rendered in dot notation, so I guess it can be skipped altogether; any other non `exp.JSONPathKey` expression can raise unsupported.

Reply:
ah yea sonuds good